corehq/apps/api/domain_metadata.py

```diff
@@ -30,6 +32,21 @@ def get_truth(inp, relate, cut):
         return ops[relate](inp, cut_datetime)
 
 
+class DomainQuerySetAdapter(object):
+
+    def __init__(self, es_query):
+        self.es_query = es_query
+
+    def count(self):
+        return self.es_query.size(0).run().total
+
+    def __getitem__(self, item):
+        if isinstance(item, slice):
+            doc_ids = self.es_query.start(item.start).size(item.stop - item.start).run().doc_ids
+            return map(Domain.wrap, iter_docs(Domain.get_db(), doc_ids))
```

Comment:
@snopoke Rest of logic in api uses domain objects e.g. https://github.com/dimagi/commcare-hq/pull/8156/files#diff-3331173c6439f2ca00e3adef54c7ced6R58. I can refactor whole api to use elasticsearch if it's worth to do it.